test/UnitTest/Microsoft.AspNet.OData.Test.Shared/DateAndTimeOfDayTest.cs

```diff
@@ -177,6 +177,7 @@ public async Task PostEntity_WithDateAndTimeOfDayProperties()
             HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Post, Uri);
 
             request.Content = new StringContent(Payload);
+            request.Content.Headers.ContentLength = Payload.Length;
```

Comment:
>ContentLength = Payload.Length; [](start = 36, length = 31)

Is Length setting required?  Why?

Reply:
I use the content length to determine whether the request has a body. So a controller action like 
```c#
Post(Person person)
{
  // ...
}
```
will only be matched if the request has a body, otherwise a 404 error is thrown. Because these parameters are not stored in the `RouteData`, this was the only way I could reasonably get such routes to be matched correctly without affecting other routes. A more detailed explanation of the criteria I added to the ODataActionSelector can be found in the second comment of this PR, as well as in [the comment on this link](https://github.com/OData/WebApi/pull/1988#issuecomment-567586566).

Reply:
From the Http Spec, content length is optional for Http request. and the Length must be zero in Get and Delete request. In the codes, if it's not required in previous version, now, it's required. That's a breaking change.

Reply:
When you say the Length must be zero in Get and Delete, does that mean that the Length header is not optional in Get and Delete?

Anyway, the reason I check the content length was just to check whether the request has a body or not. If there's a reliable way to find out whether the request body is not empty without using the content length or content type, then I can use that method. Initially I tried to check the length of the stream, but this is not reliable because the stream is not always seekable. The options is to read the stream and copy it, but I don't think this is worth it, it seems too wasteful.